### PR TITLE
Changes to match rfc7233

### PR DESCRIPTION
**Pull request by trafficone**
Addresses issue #49 - adjustment of range headers.

* feat: Added Range: Tests

Tests to evaluate
- ranges exceeding file size
- negative ranges (individually, as group, overlapping)
- RangeNotSatisfiable on range start > file size (individual, and as a group, and without a range end)
- RangeNotSatisfiable on range start < 0 (when suffix > file size)

* fix: ranges logic update

The ranges calculation in parse_range now will return values for prefix, suffix, and over-large ranges.
The rest of the logic should work with the change.

* fix: response tests pass
- Other response tests modified to match expected functionality.
- Some minor changes to websockets to allow those tests to pass as well.

### CODE REVIEW

File: tests/test_asgi.py [hunk 1]
```diff
@@ -878,7 +883,8 @@ async def app(websocket: WebSocket) -> None:
 
     client = TestClient(app)
     with pytest.raises(WebSocketDisconnect) as exc:
-        client.websocket_connect("/")
+        with client.websocket_connect("/") as _:
```
Comment (abersheeran) [inline tests/test_asgi.py:886]: Why not delete `as _`?

File: baize/asgi/websocket.py [hunk 2]
```diff
@@ -8,8 +8,9 @@
 
 
 class WebSocketDisconnect(Exception):
-    def __init__(self, code: int = 1000) -> None:
```
Comment (abersheeran) [inline baize/asgi/websocket.py:11]: This doesn't seem to be related to Range Request? If you have an idea for modifying this object, we can open another PR to discuss the modification.

File: baize/responses.py [hunk 3]
```diff
@@ -178,16 +178,19 @@ def parse_range(
             raise MalformedRangeHeader("Only support bytes range")
 
         ranges = [
-            (int(_[0]), int(_[1]) + 1 if _[1] else max_size)
-            for _ in re.findall(r"(\d+)-(\d*)", ranges_str)
+            (
+                int(_[0]) if _[0] else max_size - int(_[1]),
```
Comment (abersheeran) [inline baize/responses.py:182]: 🤔 Why use `max_size - int(_[1])`?

File: baize/responses.py [hunk 4]
```diff
@@ -178,16 +178,19 @@ def parse_range(
...
+                int(_[1]) + 1 if _[0] and _[1] and int(_[1]) < max_size else max_size,
```
Comment (abersheeran) [inline baize/responses.py:183]: Parsing `-` will got an error……We should fix the regex or judgement.

File: baize/asgi/websocket.py (hunk 2 above)
Comment (trafficone) [reply baize/asgi/websocket.py:11]: I might have been mistaken, but I thought a test was failing because it was trying to pass something to WebSocketDisconnect. I can remove the change.

File: baize/responses.py (hunk 3 above)
Comment (trafficone) [reply baize/responses.py:182]: This is to handle the suffix case. 
`Range: bytes=-10` should return the last 10 bytes of the file. The start will be the suffix distance from the end of the file, which the regex captures as _[1].

File: baize/responses.py (hunk 4 above)
Comment (trafficone) [reply baize/responses.py:183]: Good catch, I hadn't considered that case. I believe the correct response would be to return a `MalformedRangeHeader`. I will address this.

Comment (trafficone) [reply baize/responses.py:183]: Looking at the existing logic and reviewing the RFC, I think the correct answer may be to ignore it. It already ignores any section of the header that doesn't match the regex (e.g. `0-10,hello world` is valid for 1-10).

However, I think that a case where parse_range would return an empty range should result in a `MalformedRangeHeader`.

File: tests/test_asgi.py (hunk 1 above)
Comment (trafficone) [reply tests/test_asgi.py]: This was also related to the websocket code. I removed the change and everything tested fine.

**Comment by abersheeran** [comment]
Thanks！
